osu.Game/Overlays/Mods/ModSelectScreen.cs

```diff
@@ -341,6 +358,75 @@ protected override void Update()
             }
         }
 
+        private class ColumnDimContainer : Container
+        {
+            public ModColumn Column { get; }
+
+            private readonly ScrollContainer<Drawable> parentScroll;
+            private readonly Bindable<bool> isFullyOnScreen = new BindableBool(true);
+
+            [Resolved]
+            private OsuColour colours { get; set; } = null!;
+
+            public ColumnDimContainer(ScrollContainer<Drawable> parentScroll, ModColumn column)
+            {
+                this.parentScroll = parentScroll;
+                Child = Column = column;
+                column.Active.BindTo(isFullyOnScreen);
+            }
+
+            protected override void LoadComplete()
+            {
+                base.LoadComplete();
+                isFullyOnScreen.BindValueChanged(_ => updateDim(), true);
+                FinishTransforms();
+            }
+
+            private void updateDim()
+            {
+                Colour4 targetColour;
+
+                if (isFullyOnScreen.Value)
+                    targetColour = Colour4.White;
+                else
+                    targetColour = IsHovered ? colours.GrayC : colours.Gray8;
+
+                this.FadeColour(targetColour, 300, Easing.OutQuint);
+            }
+
+            protected override void Update()
+            {
+                // DrawWidth/DrawPosition do not include shear effects, and we want to know the full extents of the columns post-shear,
+                // so we have to manually compensate.
+                var topLeft = ToSpaceOfOtherDrawable(new Vector2(-DrawHeight * SHEAR, 0), parentScroll);
+                var topRight = ToSpaceOfOtherDrawable(new Vector2(DrawWidth, 0), parentScroll);
+
+                isFullyOnScreen.Value = Precision.AlmostBigger(topLeft.X, 0)
+                                        && Precision.DefinitelyBigger(parentScroll.DrawWidth, topRight.X);
+            }
+
+            protected override bool OnMouseDown(MouseDownEvent e)
```

Comment:
I'd hope this is `OnClick` instead. This would allow drag scrolling to still work (ie. dragging on the header of a column) but also feel more correct (my own expectation is that the jump doesn't occur until releasing the mouse button).